### PR TITLE
feat(Kittensgame): add presence

### LINKED ISSUE #8126: Kittens Game | kittensgame.com
### Website name

Kittens Game

### Website URL

https://kittensgame.com/web/

### Website logo

https://kittensgame.com/web/res/favicon.gif

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

A web game of building a village of cats.

Better, show the current in-game year and day in the presence.

### PR DESCRIPTION
**Pull request by DarkVillager**
Resolves #8126 
## Description 

Add presence
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/K/Kittensgame/presence.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+const presence = new Presence({
+		clientId: "1218931024592113744",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
```
Comment (DarkVillager) [inline websites/K/Kittensgame/presence.ts:6]:
```suggestion
const enum Assets {
```

File: websites/K/Kittensgame/presence.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, href, hostname } = document.location;
+
+	switch (hostname) {
+		case "kittensgame.com": {
+			const loadingBar = document.querySelector<HTMLProgressElement>(
+				"#loadingProgressBar"
+			);
```
Comment (DarkVillager) [inline websites/K/Kittensgame/presence.ts:23]:
```suggestion
```

File: websites/K/Kittensgame/presence.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+			if (loadingBar.value !== 100) {
```
Comment (DarkVillager) [inline websites/K/Kittensgame/presence.ts:24]:
```suggestion
			if (document.querySelector<HTMLProgressElement>(
				"#loadingProgressBar"
			).value !== 100) {
```

File: websites/K/Kittensgame/metadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+		"en": "Kittens Game by bloodrizer is a text-based semi-idle incremental village management simulation browser game. You manage a village of kittens as they acquire resources and unlock new technologies. The game is currently under development, so prices and balancing are subject to change."
+	},
+	"url": [
+		"kittensgame.com",
+		"forum.kittensgame.com",
+		"wiki.kittensgame.com"
+	],
+	"matches": [
+		"*://*.kittensgame.com/"
+	],
```
Comment (theusaf) [inline websites/K/Kittensgame/metadata.json:18]:
```suggestion
	"matches": [
		"*://kittensgame.com/"
		"*://*.kittensgame.com/"
	],
```

File: websites/K/Kittensgame/presence.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
+const presence = new Presence({
+		clientId: "1218931024592113744",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Advice = "https://i.imgur.com/f5RrPFn.png",
+	Logo = "https://i.imgur.com/ZnH9imj.png",
+	Loading = "https://i.imgur.com/9sInqXl.gif",
```
Comment (theusaf) [inline websites/K/Kittensgame/presence.ts:9]: @Bas950 Do .gif files still have trouble when uploaded?
```suggestion
	Loading = "https://i.imgur.com/9sInqXl\u002egif",
```

File: websites/K/Kittensgame/presence.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, href, hostname } = document.location;
+
+	switch (hostname) {
+		case "kittensgame.com": {
+			if (document.querySelector<HTMLProgressElement>(
+				"#loadingProgressBar"
+			).value !== 100) {
+				presenceData.details = "Loading..";
```
Comment (theusaf) [inline websites/K/Kittensgame/presence.ts:24]:
```suggestion
				presenceData.details = "Loading";
```
```suggestion
				presenceData.details = "Loading...";
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/K/Kittensgame/presence.ts (hunk 5 above)
Comment (Bas950) [reply websites/K/Kittensgame/presence.ts:9]: I think so, I can manually upload it if needed.

File: websites/K/Kittensgame/metadata.json [hunk 7]
```diff
@@ -14,6 +14,7 @@
 		"wiki.kittensgame.com"
 	],
 	"matches": [
+		"*://kittensgame.com/"
```
Comment (DarkVillager) [inline websites/K/Kittensgame/metadata.json:17]:
```suggestion
		"*://kittensgame.com/",
```

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**